### PR TITLE
add error states in form-behavior

### PR DESCRIPTION
**Pull request by jim-benson**
Task: task-[860065](https://ceapex.visualstudio.com/Engineering/_workitems/edit/860065)

Link: preview-[552](https://design.learn.microsoft.com/pulls/552/components/form.html)

Add error states on submit for 401, 403, and 429 errors

## Testing

1. Step one
2. Step two
   Expected result:

## Additional information

[Optional]

## Contributor checklist

- [ ] Did you update the description of this pull request with a review link and test steps?
- [ ] Did you submit a changeset? Changesets are required for all code changes.
- [ ] Does your pull request implement complex UI logic (js/ts)? Consider adding an integration test to test your user flow.
- [ ] Does your pull request add a new page to the documentation site? Add your new page for automated accessibility testing in /integration/tests/accessibility.

### CODE REVIEW

File: .changeset/shaggy-ladybugs-refuse.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@microsoft/atlas-js': minor
+---
+
+Add error states in form-behavior
```
Comment (homing1) [inline .changeset/shaggy-ladybugs-refuse.md:5]: I think the changelog message can be a bit more specific. Perhaps something like "Add handling of form submission error state in form behavior"?

File: js/src/elements/form-behavior.ts [hunk 2]
```diff
@@ -255,9 +255,18 @@ export class FormBehaviorElement extends HTMLElement {
 				const errorText = document.createElement('li');
 				errorText.innerText = this.locStrings.weEncounteredAnUnexpectedError;
 				// custom text for version mismatch
+				if (response.status === 401) {
```
[the remark this answers is not in the document]
Comment (homing1) [reply js/src/elements/form-behavior.ts:258]: Here's the preview link for your changes https://design.learn.microsoft.com/pulls/552/components/form.html

[the remark this answers is not in the document]
Comment (jim-benson) [reply js/src/elements/form-behavior.ts:263]: updated

File: js/src/elements/form-behavior.ts [hunk 3]
```diff
@@ -5,8 +5,13 @@ export const defaultMessageStrings = {
 	inputMaxLength: '{inputLabel} cannot be longer than {maxLength} characters.',
 	inputMinLength: '{inputLabel} must be at least {minLength} characters.',
 	inputRequired: '{inputLabel} is required.',
+	notAuthenticated:
+		'You are not authenticated, please refresh the page and try again. If this issue persists please log out and log back in.',
```
Comment (BenDMyers) [inline js/src/elements/form-behavior.ts:9]: I think the first sentence needs to be split into two, and a comma should be added to the final sentence:
```suggestion
		'You are not authenticated. Please refresh the page and try again. If this issue persists, please log out and log back in.',
```

File: js/src/elements/form-behavior.ts [hunk 4]
```diff
@@ -5,8 +5,13 @@ export const defaultMessageStrings = {
...
+	notAuthorized:
+		'You are not authorized to make this response, if you believe this to be in error please refresh the page and try again.',
```
Comment (BenDMyers) [inline js/src/elements/form-behavior.ts:11]: This should also be split:
```suggestion
		'You are not authorized to make this response. If you believe this to be in error, please refresh the page and try again.',
```

File: js/src/elements/form-behavior.ts [hunk 5]
```diff
@@ -5,8 +5,13 @@ export const defaultMessageStrings = {
...
 	pleaseFixTheFollowingIssues: 'Please fix the following issues to continue:',
 	thereAreNoEditsToSubmit: 'There are no edits to submit.',
+	tooManyRequests: 'You have sent too many requests, please wait a few minutes and try again.',
```
Comment (BenDMyers) [inline js/src/elements/form-behavior.ts:14]: This should also be split:

```suggestion
	tooManyRequests: 'You have sent too many requests. Please wait a few minutes and try again.',
```

File: site/src/components/form.md [hunk 6]
```diff
@@ -110,8 +110,11 @@ If you want to skip the basic validation on an input, apply a `data-skip-validat
 		loc-input-max-length="{inputLabel} cannot be longer than {maxLength} characters."
 		loc-input-min-length="{inputLabel} must be at least {minLength} characters."
 		loc-input-required="{inputLabel} is required."
+		loc-not-authenicated="You are not authenticated, please refresh the page and try again. If this issue persists please log out and log back in."
```
Comment (BenDMyers) [inline site/src/components/form.md:113]: This should be `loc-not-authenticated` (it's missing a `t`), and these sentences should be split.

```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists, please log out and log back in."
```

File: site/src/components/form.md [hunk 7]
```diff
@@ -110,8 +110,11 @@ If you want to skip the basic validation on an input, apply a `data-skip-validat
...
+		loc-not-authorized="You are not authorized to make this response, if you believe this to be in error please refresh the page and try again."
```
Comment (BenDMyers) [inline site/src/components/form.md:114]:
```suggestion
		loc-not-authorized="You are not authorized to make this response. If you believe this to be in error, please refresh the page and try again."
```

File: site/src/components/form.md [hunk 8]
```diff
@@ -110,8 +110,11 @@ If you want to skip the basic validation on an input, apply a `data-skip-validat
...
 		loc-please-fix-the-following-issues="Please fix the following issues to continue:"
 		loc-there-are-no-edits-to-submit="There are no edits to submit."
+		loc-too-many-requests="You have sent too many requests, please wait a few minutes and try again."
```
Comment (BenDMyers) [inline site/src/components/form.md:117]:
```suggestion
		loc-too-many-requests="You have sent too many requests. Please wait a few minutes and try again."
```

File: site/src/components/form.md [hunk 9]
```diff
@@ -241,8 +244,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
 		loc-input-max-length="{inputLabel} cannot be longer than {maxLength} characters."
 		loc-input-min-length="{inputLabel} must be at least {minLength} characters."
 		loc-input-required="{inputLabel} is required."
+		loc-not-authenicated="You are not authenticated, please refresh the page and try again. If this issue persists please log out and log back in."
```
Comment (BenDMyers) [inline site/src/components/form.md:247]:
```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists, please log out and log back in."
```

File: site/src/components/form.md [hunk 10]
```diff
@@ -241,8 +244,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
+		loc-not-authorized="You are not authorized to make this response, if you believe this to be in error please refresh the page and try again."
```
Comment (BenDMyers) [inline site/src/components/form.md:248]:
```suggestion
		loc-not-authorized="You are not authorized to make this response. If you believe this to be in error, please refresh the page and try again."
```

File: site/src/components/form.md [hunk 11]
```diff
@@ -241,8 +244,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
 		loc-please-fix-the-following-issues="Please fix the following issues to continue:"
 		loc-there-are-no-edits-to-submit="There are no edits to submit."
+		loc-too-many-requests="You have sent too many requests, please wait a few minutes and try again."
```
Comment (BenDMyers) [inline site/src/components/form.md:251]:
```suggestion
		loc-too-many-requests="You have sent too many requests. Please wait a few minutes and try again."
```

File: site/src/components/form.md [hunk 12]
```diff
@@ -298,8 +304,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
 		loc-input-max-length="{inputLabel} cannot be longer than {maxLength} characters."
 		loc-input-min-length="{inputLabel} must be at least {minLength} characters."
 		loc-input-required="{inputLabel} is required."
+		loc-not-authenicated="You are not authenticated, please refresh the page and try again. If this issue persists please log out and log back in."
```
Comment (BenDMyers) [inline site/src/components/form.md:307]:
```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists, please log out and log back in."
```

File: site/src/components/form.md [hunk 13]
```diff
@@ -298,8 +304,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
+		loc-not-authorized="You are not authorized to make this response, if you believe this to be in error please refresh the page and try again."
```
Comment (BenDMyers) [inline site/src/components/form.md:308]:
```suggestion
		loc-not-authorized="You are not authorized to make this response. If you believe this to be in error, please refresh the page and try again."
```

File: site/src/components/form.md [hunk 14]
```diff
@@ -298,8 +304,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
 		loc-please-fix-the-following-issues="Please fix the following issues to continue:"
 		loc-there-are-no-edits-to-submit="There are no edits to submit."
+		loc-too-many-requests="You have sent too many requests, please wait a few minutes and try again."
```
Comment (BenDMyers) [inline site/src/components/form.md:311]:
```suggestion
		loc-too-many-requests="You have sent too many requests. Please wait a few minutes and try again."
```

**Review by BenDMyers — CHANGES_REQUESTED** [review]

File: site/src/components/form.md [hunk 15]
```diff
@@ -110,8 +110,11 @@ If you want to skip the basic validation on an input, apply a `data-skip-validat
 		loc-input-max-length="{inputLabel} cannot be longer than {maxLength} characters."
 		loc-input-min-length="{inputLabel} must be at least {minLength} characters."
 		loc-input-required="{inputLabel} is required."
+		loc-not-authenicated="You are not authenticated. Please refresh the page and try again. If this issue persists please log out and log back in."
```
Comment (BenDMyers) [inline site/src/components/form.md:113]: This still has a typo in the attribute name:

```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists please log out and log back in."
```

File: site/src/components/form.md [hunk 16]
```diff
@@ -298,8 +304,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
```
Comment (BenDMyers) [inline site/src/components/form.md:307]:
```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists please log out and log back in."
```

**Review by BenDMyers — CHANGES_REQUESTED** [review]

**Review by homing1 — CHANGES_REQUESTED** [review]

File: site/src/components/form.md [hunk 17]
```diff
@@ -110,8 +110,11 @@ If you want to skip the basic validation on an input, apply a `data-skip-validat
 		loc-input-max-length="{inputLabel} cannot be longer than {maxLength} characters."
 		loc-input-min-length="{inputLabel} must be at least {minLength} characters."
 		loc-input-required="{inputLabel} is required."
+		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists please log out and log back in."
+		loc-not-authorized="You are not authorized to make this response. If you believe this to be in error please refresh the page and try again."
```
Comment (homing1) [inline site/src/components/form.md:114]:
```suggestion
"You are not authorized to make this response. If you believe this to be in error, please refresh the page and try again."
```

File: site/src/components/form.md [hunk 18]
```diff
@@ -241,8 +244,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
```
Comment (homing1) [inline site/src/components/form.md:248]:
```suggestion
"You are not authorized to make this response. If you believe this to be in error, please refresh the page and try again."
```

File: site/src/components/form.md [hunk 19]
```diff
@@ -298,8 +304,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
```
Comment (homing1) [inline site/src/components/form.md:308]:
```suggestion
"You are not authorized to make this response. If you believe this to be in error, please refresh the page and try again."
```

**Review by homing1 — CHANGES_REQUESTED** [review]

File: site/src/components/form.md [hunk 20]
```diff
@@ -110,8 +110,11 @@ If you want to skip the basic validation on an input, apply a `data-skip-validat
 		loc-input-max-length="{inputLabel} cannot be longer than {maxLength} characters."
 		loc-input-min-length="{inputLabel} must be at least {minLength} characters."
 		loc-input-required="{inputLabel} is required."
+		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists please log out and log back in."
```
Comment (BenDMyers) [inline site/src/components/form.md:113]: nit:
```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists, please log out and log back in."
```

File: site/src/components/form.md [hunk 21]
```diff
@@ -241,8 +244,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
```
Comment (BenDMyers) [inline site/src/components/form.md:247]: nit:
```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists, please log out and log back in."
```

File: site/src/components/form.md [hunk 22]
```diff
@@ -298,8 +304,11 @@ If there is a need to hide the validation banner on top of the form, we can appl
...
```
Comment (BenDMyers) [inline site/src/components/form.md:307]: nit:
```suggestion
		loc-not-authenticated="You are not authenticated. Please refresh the page and try again. If this issue persists, please log out and log back in."
```

**Comment by BenDMyers** [comment]
My bad on the recent nits; GitHub was slow to update the previews!

**Review by jim-benson — COMMENTED** [review]
updated

**Review by jim-benson — COMMENTED** [review]
updated

**Review by homing1 — APPROVED** [review]

**Pull request merged**